dts/common/nordic/nrf54h20.dtsi

```diff
@@ -93,6 +93,12 @@
 			#clock-cells = <0>;
 			clock-frequency = <DT_FREQ_M(16)>;
 		};
+
+		exmif_hsfll: exmif_hsfll {
+			compatible = "fixed-clock";
+			#clock-cells = <0>;
+			clock-frequency = <DT_FREQ_M(400)>;
+		};
```

Comment:
note: commit message says app needs to turn this on/off, so this is not really a fixed-clock, it will require a clock control driver that uses nrfs underneath. Looks like we'll need a new compatible, eg `nordic,nrf-hsfll-remote` or similar and a driver so we can use `clock_control_on/off`. @anangl any comments here?

Reply:
Should this be a part of this PR, or could this be addressed later?

Reply:
AFAICT the driver only supports the fixed clock frequency now: https://github.com/zephyrproject-rtos/zephyr/blob/0b87cb1545d5ec99b4f5e8368bb04dfbee4098b1/drivers/spi/spi_dw.c#L636-L641 so I think there is no point in trying to provide proper handling of that HSFLL here. I think the initial approach with the `clock-frequency` property in the `exmif` node was fine. Especially taking into account that the main idea behind this PR is to just take advantage of the already existing `spi_dw` driver to use the EXMIF peripheral in its basic SPI mode, temporarily, until we have a proper EXMIF driver.

Reply:
Looking at the above snippet the previous approach with `clock-frequency` gets the job done just the same as `clocks`. I'll stay with the current approach (`clocks`) unless some problem emerges.

Reply:
Yes, it does the job in the `spi_dw` driver context, but the HSFLL is not actually a fixed clock, as its output frequency can be changed, so the added node is a bit misleading.

Reply:
Okay, I restored it to the previous approach.